api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2620,6 +2643,52 @@ public void removeObject(String bucketName, String objectName)
     executeDelete(bucketName, objectName, null);
   }
 
+  /**
+   * Removes an object if object is locked by Governance Mode
+   *
+   * <pre>Example:{@code
+   * minioClient.removeObject("my-bucketname", deleteobject, true);
+   * }</pre>
+   *
+   * @param bucketName Name of the bucket.
+   * @param deleteObject DeleteObject consisting of name and versionId.
+   * @param bypassGovernanceRetention Specifies whether you want to delete this object even if it
+   *     has a Governance-type Object Lock in place.
+   * @throws ErrorResponseException thrown to indicate S3 service returned an error response.
+   * @throws IllegalArgumentException throws to indicate invalid argument passed.
+   * @throws InsufficientDataException thrown to indicate not enough data available in InputStream.
+   * @throws InternalException thrown to indicate internal library error.
+   * @throws InvalidBucketNameException thrown to indicate invalid bucket name passed.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws InvalidResponseException thrown to indicate S3 service returned invalid or no error
+   *     response.
+   * @throws IOException thrown to indicate I/O error on S3 operation.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws XmlParserException thrown to indicate XML parsing error.
+   */
+  public void removeObject(
+      String bucketName, DeleteObject deleteObject, boolean bypassGovernanceRetention)
```

Comment:
Many S3 APIs supported version ID. An object is a referred as `name` and `version ID` optionally,  My recommendation is to introduce `Object` class containing `name` and `versionId` as separate PR.

Its bad idea to have `DeleteObject` class. Here you just introduce `bypassGovernanceRetention` in single object delete and multi-object delete and limit this PR scope.